src/HealthChecks.Rabbitmq/RabbitMQHealthCheck.cs

```diff
@@ -49,47 +53,12 @@ public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, Canc
 
     private IConnection EnsureConnection()
     {
-        if (_disposed)
-            throw new ObjectDisposedException(nameof(RabbitMQHealthCheck));
-
         if (_connection == null)
         {
-            if (_factory == null)
-            {
-                _factory = new ConnectionFactory()
-                {
-                    Uri = _rabbitConnectionString,
-                    AutomaticRecoveryEnabled = true,
-                    UseBackgroundThreadsForIO = true,
-                };
-
-                if (_sslOption != null)
-                    ((ConnectionFactory)_factory).Ssl = _sslOption;
-            }
-
-            _connection = _factory.CreateConnection();
+            Guard.ThrowIfNull(_factory);
+            _connection = _connections.GetOrAdd(_factory.Uri.ToString(), _ => _factory.CreateConnection());
```

Comment:
`SslOption` should be included in key

Reply:
This cannot be done for the `RabbitMQHealthCheck(IConnectionFactory)` constructor: `IConnectionFactory` does not expose SSL options (while `ConnectionFactory` does). One solution could be use the hash: `_factory.GetHashCode()` as key.

Reply:
My concern is about doing something like that:
```c#
.AddRabbitMQ(sp => "amqp://localhost:6672/")
.AddRabbitMQ(sp => "amqp://localhost:6672/", new SslOption { ServerName = "srv", CertPath = "..." })
```

The same uri but different ssl settings. Now the second registration effectively discards ssl options.

Reply:
Yes I understood. Even if I don't see any clear use case of doing this, I think we should allow it. For the moment I cannot think nothing better than using the factory itself (`_facotry.GetHashCode()`) as key... but I not really found of it.

Reply:
Let's take timeout to find solution.

Reply:
For RavenDB HC the options object is currently used as key: https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/2806c8033352077cda1f2fad5be7dbcf06814637/src/HealthChecks.RavenDB/RavenDBHealthCheck.cs#L28
So using the connection factory as key would be no new concept to this library

Reply:
So... let's introduce `RabbitMQHealthCheckOptions`. I am inclined to use options in all checks at least for those that I changed.